astropy/coordinates/baseframe.py

```diff
@@ -1581,6 +1581,36 @@ def __setattr__(self, attr, value):
 
         super().__setattr__(attr, value)
 
+    def __eq__(self, value):
+        """Equality operator for frame.
+
+        This implements strict equality and requires that the frames are
+        equivalent and that the representation data are exactly equal.
+        """
+        if not self.is_equivalent_frame(value):
+            return False
+
+        if value._data is None and self._data is not None:
+            return False
+
+        if self._data is None and value._data is not None:
+            return False
+
+        if self._data is None and value._data is None:
+            return True
+
+        if self.shape != value.shape:
```

Comment:
Independently of my larger comment, I would allow different shapes (but catch the possible broadcast error and return `False`), i.e., `sc[0] == sc` should be possible.